addons/mail/models/mail_channel.py

```diff
@@ -238,12 +376,53 @@ def message_get_recipient_values(self, notif_message=None, recipient_ids=None):
             }
         return super(Channel, self).message_get_recipient_values(notif_message=notif_message, recipient_ids=recipient_ids)
 
+    def _extract_values(self, message_type, **kwargs):
+        """ This method is used to compute moderation status before the creation of a message. 
+            For this operation the message's author email address is required.
+            This address is returned with status for other computations.
+        """
+        moderation_status = 'accepted'
+        email = ''
+        if self.moderation and message_type in ['email', 'comment']:
+            author_id = kwargs.get('author_id')
+            if author_id and isinstance(author_id, pycompat.integer_types):
+                author = self.env['res.partner'].browse([author_id])
+                email = author.email
+            elif author_id:
+                email = author_id.email
+            elif self.env.user:
+                email = self.env.user.email
+            else:
+                email_from = kwargs.get('email_from')
+                if email_from:
+                    email = tools.email_split(email_from)[0]
+            if self.env['mail.moderation.email'].search([('email', '=', email), ('status', '=', 'allow')]) or email in self.mapped('moderator_ids.email'):
```

Comment:
Why do you make a global search on mail.mod.email, not limited to the current channel ? Is it intended or an issue? Normally ban / allow is done per channel if I understood well.

Reply:
Thanks! It was a mistake.